packages/medusa/src/api-v2/admin/products/route.ts

```diff
@@ -1,22 +1,58 @@
+import { createProductsWorkflow } from "@medusajs/core-flows"
+import { CreateProductDTO } from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  isString,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
 import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "../../../types/routing"
-
-import { CreateProductDTO } from "@medusajs/types"
-import { createProductsWorkflow } from "@medusajs/core-flows"
-import { remoteQueryObjectFromString } from "@medusajs/utils"
+import { listPriceLists } from "../price-lists/queries"
 
 export const GET = async (
   req: AuthenticatedMedusaRequest,
   res: MedusaResponse
 ) => {
-  const remoteQuery = req.scope.resolve("remoteQuery")
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const filterableFields = { ...req.filterableFields }
+  const filterByPriceListIds =
+    (filterableFields.price_list_id as string[]) || []
+  const priceListVariantIds: string[] = []
+
+  // When filtering by price_list_id, we need use the remote query to get
+  // the variant IDs through the price list price sets.
+  if (filterByPriceListIds.length) {
+    const [priceLists] = await listPriceLists({
+      container: req.scope,
+      fields: ["prices.variant_id"],
+      variables: { filters: { id: filterByPriceListIds }, skip: 0, take: null },
```

Comment:
Not doing pagination might explode here, we need to think how cases like this should be handled. Can you just add a TODO comment maybe?

Reply:
can you elaborate on why this might explode?

Reply:
Agree on that. We def shouldn't be removing pagination on unscoped entities. This will only return as many price_lists as there are filter of price_list_ids here. The take:null here is more to not return a false result in case someone ever tried to scope price_list_id greater than the default pagination limit (which in itself is an edge case scenario).